detection_rules/version_lock.py

```diff
@@ -43,25 +105,40 @@ def __init__(self, version_lock_file: Optional[str] = None, deprecated_lock_file
         self.version_lock_file = version_lock_file
         self.deprecated_lock_file = deprecated_lock_file
 
-        self.version_lock = load_etc_dump(self.version_lock_file) if version_lock_file else version_lock
-        self.deprecated_lock = load_etc_dump(self.deprecated_lock_file) if deprecated_lock_file else deprecated_lock
+        if version_lock_file:
+            self.version_lock = VersionLockFile.load_from_file(version_lock_file)
+        else:
+            self.version_lock = VersionLockFile.from_dict(version_lock)
 
-    def save_versions(self, current_versions: dict):
-        save_etc_dump(current_versions, self.version_lock_file)
-        print('Updated version.lock.json file')
+        if deprecated_lock_file:
+            self.deprecated_lock = DeprecatedRulesFile.load_from_file(deprecated_lock_file)
+        else:
+            self.deprecated_lock = DeprecatedRulesFile.from_dict(deprecated_lock)
+
+    @staticmethod
+    def save_file(path: Path, lock_file: Union[VersionLockFile, DeprecatedRulesFile]):
+        assert path, f'{path} not set'
+        lock_file.save_to_file(path)
+        print(f'Updated {path} file')
 
     def get_locked_version(self, rule_id: str, min_stack_version: Optional[str] = None) -> Optional[int]:
         if rule_id in self.version_lock:
             latest_version_info = self.version_lock[rule_id]
-            stack_version_info = latest_version_info.get("previous", {}).get(min_stack_version, latest_version_info)
-            return stack_version_info['version']
+            if latest_version_info.previous and latest_version_info.previous.get(min_stack_version):
+                stack_version_info = latest_version_info.previous.get(min_stack_version)
+            else:
+                stack_version_info = latest_version_info
+            return stack_version_info.version
 
     def get_locked_hash(self, rule_id: str, min_stack_version: Optional[str] = None) -> Optional[str]:
         """Get the version info matching the min_stack_version if present."""
         if rule_id in self.version_lock:
             latest_version_info = self.version_lock[rule_id]
-            stack_version_info = latest_version_info.get("previous", {}).get(min_stack_version, latest_version_info)
-            existing_sha256: str = stack_version_info['sha256']
+            if latest_version_info.previous and latest_version_info.previous.get(min_stack_version):
+                stack_version_info = latest_version_info.previous.get(min_stack_version)
+            else:
+                stack_version_info = latest_version_info
+            existing_sha256: str = stack_version_info.sha256
             return existing_sha256
 
     def manage_versions(self, rules: RuleCollection,
```

Comment:
```suggestion
    def manage_versions(self, rules: RuleCollection,
                        exclude_version_update=False, save_changes=False,
                        verbose=True) -> Tuple[List[str], List[str], List[str]]:
```

Reply:
I think we can be implicit with tuple hinting here 🙂 . 3.9 introduced typing with [generics](https://docs.python.org/3/whatsnew/3.9.html#type-hinting-generics-in-standard-collections) 🎉 , so this feels consistent with that

Reply:
Nice. It looks like importing from `typing` is deprecated as of 3.9. In that case we should be consistent. Either support explicitly for 3.8 or 3.9+ and use the generics. 

So that we don't scope creep, we can just make this one line consistent to use either, and create a new issue to make typing consistent.

Reply:
yea we can punt - I don't think we need a complete typing revamp until we get to a point it is required though 👍